include/solvers/mpm_base.tcc

```diff
@@ -329,6 +329,36 @@ bool mpm::MPMBase<Tdim>::initialise_particles() {
                        particles_volume_end - particles_volume_begin)
                        .count());
 
+    // Get material sets
+    bool update_materials = false;
+    try {
+      update_materials = mesh_props["update_materials"].template get<bool>();
```

Comment:
Why do we need a separate `update_materials` flag? Can we not just use the presence of `material_sets` to overwrite assigned materials?

Reply:
Removed `bool` flag